### ISSUE TITLE
Terraform crashes and hangs on plan - no crash.log is reproduced unless validate is ran

### ISSUE THREAD

**Issue by jar145** [hashicorp/terraform#24061]
### Terraform Version

```
Terraform v0.12.20
```

### Terraform Configuration Files

This is a complex private repo and not sure if I can post everything here (not sure how to do the GPG key encryption either), but here is the provider I am using. I am also using privately made modules as well, but as far as I can tell that isn't the issue as to the best of my knowledge they are being used elsewhere.
```
provider "aws" {
  allowed_account_ids = ["xxxxxxxxxxxx"]
  region              = "us-west-2"

  version = "~> 2.30"
}

terraform {
  required_version = ">= 0.12.0"
}
```

### Debug Output

Here are the debug files for both plan and validate
https://gist.github.com/jar145/e9459d7c80bfcfa54e1ead3d51c8c5b1

### Crash Output

Gist for the crash.log file produced upon validate (plan just hangs and doesn't produce anything)
https://gist.github.com/jar145/9bf0df828b57d303609f8e7a47fe8bd7

### Expected Behavior

What should have happened?
Terraform plan/validate runs successfully or gives an error

### Actual Behavior

What actually happened?
Terraform crashes

When plan is ran, terraform hangs after crash message with no crash.log file.
When validate is ran, the terrafrom process ends after the crash and a crash.log file is produced

### Steps to Reproduce

Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform plan`

-or-
1. `terraform init`
2. `terraform validate`

### Additional Context

Nothing unusual that I am aware of, but this is a workspace that is hosted on terraform cloud

**Comment by danieldreier** [comment]
When I run this Terraform 0.12.25 with the sample code you gave I get the following output and have been unable to reproduce the failure mode you're describing:

```
terraform -version                                                                                                                                                Terraform v0.12.25
+ provider.aws v2.61.0

terraform init

Initializing the backend...

Initializing provider plugins...
- Checking for available provider plugins...
- Downloading plugin for provider "aws" (hashicorp/aws) 2.61.0...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

terraform plan

Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.


------------------------------------------------------------------------

No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.

terraform validate
Success! The configuration is valid.
```

Can you help me put together a reproduction case where it crashes like you were running into? I believe you that you had a real crash because I see the
```
panic: inconsistent map element types (cty.String then cty.Object(map[string]cty.Type{"container_definitions":cty.DynamicPseudoType, "container_name":cty.String, "container_port":cty.Number, "task_role_arn":cty.DynamicPseudoType}))
```
error in your putput. I suspect this is actually a duplicate of #24067 because it shows the same error message in the same `cty.MapVal` place. Can you please take a look at that issue, and then either help put together a more detailed reproduction case here or help decide whether this is a duplicate? My impulse is to close this out in favor of that one because I think they're the same issue.

**Comment by apparentlymart** [comment]
Hi @jar145,

We didn't hear back from you after @danieldreier's recent comment, so I'm going to close this. I agree with him that this seems to be the same root cause as #24067, and that issue has now been fixed in our main branch ready to be included in the forthcoming v0.13.0 release. If you still see this problem after v0.13.0 is released and you've been able to upgrade to it, please let us know! Thanks.
